### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by freshfab**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -146,11 +144,11 @@ func (c *Config) validateFilePVConfig() error {
...
-	if keyFile, err := os.Stat(c.FilePV.KeyFilePath); err != nil && !keyFile.IsDir() {
-		errs += "\tkey_file_path is not a valid path\n"
+	if _, err := os.Stat(c.FilePV.KeyFilePath); err != nil {
+		errs += "\tkey_file_path does not exist\n"
 	}
-	if stateFile, err := os.Stat(c.FilePV.StateFilePath); err != nil && !stateFile.IsDir() {
-		errs += "\tstate_file_path is not a valid path\n"
+	if _, err := os.Stat(c.FilePV.StateFilePath); err != nil {
+		errs += "\tstate_file_path does not exist\n"
 	}
```
Comment (dee-effes) [inline config/config.go:152]: Do we still need those checks?

File: cmd/init/config_builder_test.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+func TestBuildConfigTemplate(t *testing.T) {
+	defer os.Remove("./pairmint.toml")
+	if err := BuildConfigTemplate("."); err != nil {
+		t.Errorf("Expected err to be nil, instead got: %v", err)
+	}
+	if err := BuildConfigTemplate("/some/non/existent/path"); err == nil {
+		t.Errorf("Expected err, instead got nil")
+	}
+}
```
Comment (dee-effes) [inline cmd/init/config_builder_test.go:16]: Seems like we are testing os package functionality

File: README.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # Pairmint
```
Comment (dee-effes) [inline README.md:1]: replace pairmint with signCTRL

File: README.md [hunk 4]
```diff
@@ -1,5 +1,10 @@
...
 
+[![Go Report Card](https://goreportcard.com/badge/github.com/blockscapelab/raftify)](https://goreportcard.com/report/github.com/blockscapenetwork/pairmint)
+[![License](https://img.shields.io/github/license/cosmos/cosmos-sdk.svg)](https://github.com/cosmos/cosmos-sdk/blob/master/LICENSE)
+
+> :warning: Be mindful of key security as Pairmint currently only supports Tendermint's file-based signer. Make sure to properly secure your validator instance from unauthorized access.
```
Comment (dee-effes) [inline README.md:6]: replace pairmint with signCTRL

File: README.md [hunk 5]
```diff
@@ -1,5 +1,10 @@
...
+
 Pairmint is a high availability solution for Tendermint-based blockchain validators. It uses the blockchain itself as a perfectly synchronous communication line between redundant validators running in parallel for double-signing protection.
```
Comment (dee-effes) [inline README.md:8]: replace pairmint with signCTRL

File: README.md [hunk 6]
```diff
@@ -36,7 +41,7 @@ $ make install
 
 ## Configuration
 
-Before putting Pairmint into operation, it needs to be initialized using:
+Each validator node runs in tandem with its own Pairmint daemon, and thus each one also has its own configuration. You can initialize Pairmint using:
```
Comment (dee-effes) [inline README.md:44]: replace pairmint with signCTRL

File: README.md [hunk 7]
```diff
@@ -54,6 +59,8 @@ If you do already have a keypair, you can either copy them into the `$PAIRMINT_C
 
 Init contains configuration parameters needed on initialization.
 
+> :information_source: The configuration parameters `set_size` and `threshold` must be the same across all validator nodes in the Pairmint set.
```
Comment (dee-effes) [inline README.md:62]: replace pairmint with signCTRL

File: README.md [hunk 8]
```diff
@@ -36,7 +41,7 @@ $ make install
 
 ## Configuration
 
-Before putting Pairmint into operation, it needs to be initialized using:
+Each validator node runs in tandem with its own Pairmint daemon, and thus each one also has its own configuration. You can initialize Pairmint using:
 
 ```shell
 $ pairmint init
```
Comment (dee-effes) [inline README.md:47]: replace pairmint with signCTRL

File: README.md [hunk 9]
```diff
@@ -67,11 +74,52 @@ Init contains configuration parameters needed on initialization.
...
+| Parameter         | Type   | Description                                                                                                               |
+| :---------------- | :----- | :------------------------------------------------------------------------------------------------------------------------ |
+| `chain_id`        | string | Chain ID the validator is part of.                                                                                        |
+| `key_file_path`   | string | Path to the `priv_validator_key.json` file. Defaults to `$PAIRMINT_CONFIG_DIR/priv_validator_key.json` if left empty.     |
+| `state_file_path` | string | Path to the `priv_validator_state.json` file. Defaults to `$PAIRMINT_CONFIG_DIR/priv_validator_state.json` if left empty. |
+
+#### Example
+
+Let's assume, you have two validators in your Pairmint set. The following configurations are examples of their respective `pairmint.toml` files.
+An example of a valid `pairmint.toml` could look like this:
```
Comment (dee-effes) [inline README.md:86]: replace pairmint with signCTRL

File: README.md [hunk 10]
```diff
@@ -67,11 +74,52 @@ Init contains configuration parameters needed on initialization.
...
-| `key_file_path`   | string | Path to the `priv_validator_key.json` file. Defaults to `$PAIRMINT_CONFIG_DIR/priv_validator_key.json`.     |
-| `state_file_path` | string | Path to the `priv_validator_state.json` file. Defaults to `$PAIRMINT_CONFIG_DIR/priv_validator_state.json`. |
+| Parameter         | Type   | Description                                                                                                               |
+| :---------------- | :----- | :------------------------------------------------------------------------------------------------------------------------ |
+| `chain_id`        | string | Chain ID the validator is part of.                                                                                        |
+| `key_file_path`   | string | Path to the `priv_validator_key.json` file. Defaults to `$PAIRMINT_CONFIG_DIR/priv_validator_key.json` if left empty.     |
+| `state_file_path` | string | Path to the `priv_validator_state.json` file. Defaults to `$PAIRMINT_CONFIG_DIR/priv_validator_state.json` if left empty. |
```
Comment (dee-effes) [inline README.md:81]: replace pairmint with signCTRL

File: README.md [hunk 11]
```diff
@@ -67,11 +74,52 @@ Init contains configuration parameters needed on initialization.
...
+threshold = 10 # Must be 10 for both validators!
+rank = 1       # Must be unique! (No two validators can have the same rank)
+validator_laddr = "127.0.0.1:3000"
+validator_laddr_rpc = "127.0.0.1:26657"
+
+[file_pv]
+
+chain_id = "mychain"
+key_file_path = "/Users/myuser/.pairmint/priv_validator_key.json"
+state_file_path = "/Users/myuser/.pairmint/priv_validator_state.json"
```
Comment (dee-effes) [inline README.md:103]: replace pairmint with signCTRL

File: README.md [hunk 12]
```diff
@@ -80,5 +128,3 @@ After creating the configuration, start Pairmint using:
 ```shell
 $ pairmint start
```
Comment (dee-effes) [inline README.md:129]: replace pairmint with signCTRL

File: config/config_test.go [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+		FilePV: FilePVConfig{
+			ChainID:       "testchain",
+			KeyFilePath:   "./priv_validator_key.json",
+			StateFilePath: "./priv_validator_state.json",
+		},
+	}
+}
+
+func TestInitDir(t *testing.T) {
+	configDir := "./.pairminttest"
```
Comment (dee-effes) [inline config/config_test.go:50]: replace pairmint with signCTRL

File: config/config_test.go [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+	if err := InitDir(configDir); err != nil {
+		t.Errorf("Expected err to be nil, instead got: %v", err)
+	}
+	if _, err := os.Stat(configDir); os.IsNotExist(err) {
+		t.Errorf("Expected %v to have been created, instead it doesn't exist", configDir)
+	}
+}
+
+func TestGetDir(t *testing.T) {
+	defaultDir := os.Getenv("HOME") + "/.pairmint"
```
Comment (dee-effes) [inline config/config_test.go:62]: replace pairmint with signCTRL

File: config/config.go [hunk 15]
```diff
@@ -62,7 +62,7 @@ type Config struct {
 }
 
 // InitDir creates the pairmint configuration directory according
-// to the `PAIRMINT_CONFIG_DIR` encironment variable.
+// to the PAIRMINT_CONFIG_DIR environment variable.
```
Comment (dee-effes) [inline config/config.go:65]: replace pairmint with signCTRL

File: connection/rwc_test.go [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+			errCh <- fmt.Errorf("Expected err to be nil, instead got: %v", err)
+			return
+		}
+
+		secretConn, err := p2pconn.MakeSecretConnection(conn, tmcrypto.PrivKey(priv))
+		if err != nil {
+			errCh <- fmt.Errorf("Expected err to be nil, instead got: %v", err)
+			return
+		}
+		defer secretConn.Close()
```
Comment (dee-effes) [inline connection/rwc_test.go:44]: Couldn't this be a race condition? what if the goroutine is finished early?

File: cmd/init/config_builder_test.go (hunk 2 above)
Comment (freshfab) [reply cmd/init/config_builder_test.go:16]: Pretty much, yeah 😐  I'm reworking the config builder in the upcoming rewrite 👍🏼

File: config/config.go (hunk 1 above)
Comment (freshfab) [reply config/config.go:152]: I'd like to keep them to make sure the keys are actually there. There's really no point in starting SignCTRL if they're not there.

**Pull request merged**